lib/main_common.pm

```diff
@@ -1842,4 +1843,49 @@ sub load_sles4sap_tests {
     }
 }
 
+sub updates_is_applicable {
+    # we don't want live systems to run out of memory or virtual disk space.
+    # Applying updates on a live system would not be persistent anyway.
+    # Also, applying updates on BOOT_TO_SNAPSHOT is useless.
+    # Also, updates on INSTALLONLY do not match the meaning
+    return !get_var('INSTALLONLY') && !get_var('BOOT_TO_SNAPSHOT') && !get_var('DUALBOOT') && !get_var('UPGRADE') && !is_livesystem;
+}
+
+sub guiupdates_is_applicable {
+    return get_var("DESKTOP") =~ /gnome|kde|xfce|lxde/ && !check_var("FLAVOR", "Rescue-CD");
+}
+
+sub load_system_update_tests {
+    if (is_sle) {
+        if (x11tests_is_applicable() && gnomestep_is_applicable() && !is_staging) {
+            loadtest "update/updates_packagekit_gpk";
+        }
+        elsif (consolestep_is_applicable()) {
+            loadtest "update/zypper_up";
+        }
+    }
+    else {
+        return unless updates_is_applicable();
+        if (need_clear_repos) {
+            loadtest "update/zypper_clear_repos";
+        }
+        if (get_var('ZYPPER_ADD_REPOS')) {
+            loadtest "console/zypper_add_repos";
+        }
+        if (guiupdates_is_applicable()) {
+            loadtest "update/prepare_system_for_update_tests";
+            if (check_var("DESKTOP", "kde")) {
+                loadtest "update/updates_packagekit_kde";
+            }
+            else {
+                loadtest "update/updates_packagekit_gpk";
+            }
+            loadtest "update/check_system_is_updated";
+        }
+        else {
+            loadtest "update/zypper_up";
+        }
+    }
+}
```

Comment:
ok, this looks like it's going in the right direction but you still have the whole update function still in products/opensuse/main.pm which I think you can remove and fully replace by this one.